source/MaterialXFormat/XmlIo.cpp

```diff
@@ -106,6 +107,11 @@ void elementToXml(ConstElementPtr elem, xml_node& xmlNode, const XmlWriteOptions
             {
                 if (!writtenSourceFiles.count(sourceUri))
                 {
+                    // Check to see if we have flagged this XInclude to be ignored, if so skip it.
```

Comment:
@jstone-lucasfilm , currently we've added this in so that we can import libraries implicitly and allow for xincludes which are explicit library imports. When we write a file we want to be able to not write out implicit ones but still write out explicit ones. 

Let us know what you think of this. This could be an xinclude "write" predicate instead.

Reply:
Couldn't this functionality be handled by the existing write predicate for _elements_?  If you specified a write predicate that returned false for any element with a specific source URI, then the XInclude associated with that source URI should also be omitted from the written document.

Reply:
This is true. We can add in the element predicate or in our case add to our existing element predicate. I'll revert this modification.